lib/galaxy/util/commands.py

```diff
@@ -110,9 +111,11 @@ def argv_to_str(command_argv, quote=True):
     return " ".join(map_func(c) for c in command_argv if c is not None)
 
 
-def _wait(cmds, **popen_kwds):
+def _wait(cmds, input=None, **popen_kwds):
     p = subprocess.Popen(cmds, **popen_kwds)
-    stdout, stderr = p.communicate()
+    stdout, stderr = p.communicate(input)
+    if not six.PY2:
+        stdout, stderr = unicodify(stdout).strip(), unicodify(stderr).strip()
```

Comment:
```suggestion
    stdout, stderr = unicodify(stdout).strip(), unicodify(stderr).strip()
```
We want unicode everywhere. This is fine for python 2 as well.

Reply:
Actually I took this from here: https://github.com/galaxyproject/galaxy/blob/25f33300ab737358fc033d6e5aed00cd700a7216/lib/galaxy/util/commands.py#L42 

should I also change this?

Reply:
IMO yes, but I do not understand why we did this in the first place. Both python 2 and python 3 seem to support writing unicode to stdout/stderr. @nsoranzo any advice here ?

Reply:
I think this should be kept as it is now in the PR, i.e. unicodify only on Python3 inside `redirect_aware_commmunicate()`, unicodify always inside `_wait()`.
The redirect code was written mainly for planemo and it's safer to keep as it is.

Reply:
ok